Support hex encoding output

huylenq opened the pull request.

huylenq committed:
CheckStyle fixes

huylenq committed:
Add "format" config, supports "base64" (as default) and "hex"

huylenq committed:
Update README with for the new "format" config

dmikurube commented on src/main/java/org/embulk/filter/encrypt/EncryptFilterPlugin.java:
```diff
@@ -106,12 +106,58 @@ public String toString()
         @Config("algorithm")
         public Algorithm getAlgorithm();
 
+        @Config("format")
```
The name `format` sounds a bit different from what it does. It is a kind of encoding, not just formatting.

What do you think `output_encoding` or like that?

huylenq replied:
I do agree

dmikurube commented on README.md:
```diff
@@ -23,7 +23,8 @@ You can apply encryption to password column and get following outputs:
 - **key_hex**: encryption key (string, required)
-- **iv_hex**: encyrption initialization vector (string, required if mode of the algorithm is CBC)
+- **iv_hex**: encryption initialization vector (string, required if mode of the algorithm is CBC)
+- **format**: encoding of the encrypted value, can be either "base64" or "hex" (base16)
```
Same as the comment of mine in the code.

dmikurube commented on README.md:
```diff
@@ -120,6 +121,7 @@ filters:
     column_names: [password, ip]
     key_hex: 098F6BCD4621D373CADE4E832627B4F60A9172716AE6428409885B8B829CCB05
     iv_hex: C9DD4BB33B827EB1FBA1B16A0074D460
+    format: hex
```
ditto.

dmikurube reviewed (commented):
Basically looks good. Just commented about naming.

dmikurube reviewed (commented):
And, please add unit test(s) for this feature.

huylenq committed:
Add unit tests

huylenq commented:
Unit tests added, also cover existing implementation.

huylenq committed:
Fix: mismatched algorithm on a test case

huylenq committed:
Make tests a bit more readable

huylenq committed:
Improve a test case, make it immune to change on the reusable base config

huylenq committed:
Change the config parameter "format" to "output_encoding"

huylenq committed:
Update an example on README: `algorithm` is required

dmikurube reviewed (approved):
Looks good to me, but please confirm with @kietdo360 as well.

huylenq committed:
Make "key_hex" required, to get free validation with an appropriate message

Currently, if "key_hex" is absence, it will eventually fail anyway with a
ConfigException that is reraised from an IllegalStateException
from `Optional.get()`

huylenq committed:
Add unit tests for some expected error messages

huylenq commented:
Also changed the way `key_hex` is required for more meaningful error message. @kietdo360 could you take a look?

kietdo360 reviewed (approved):
LGTM

huylenq committed:
Change: presence of IV config on non-IV algorithm should log out warning instead of throwing exception
